src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ACTION_CLEAR = 'clear';
+  const ACTION_ADD_PROPERTIES = 'addProperties';
+  const ACTION_REMOVE__PROPERTIES = 'removeProperties';
+  const transformStateResult = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ACTION_CLEAR:
+        const keys = Object.keys(stateClone);
+
+        for (const key of keys) {
+          delete stateClone[key];
+        }
+        break;
+
+      case ACTION_ADD_PROPERTIES:
+        Object.assign(stateClone, action.extraData);
+        break;
+
+      case ACTION_REMOVE__PROPERTIES:
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+    }
```

Comment:
[CODE STYLE]: The `switch` statement lacks a `default` case. It's a good practice to include a `default` case for error handling to cover unexpected `action.type` values.